src/anemoi/graphs/plotting/interactive_html.py

```diff
@@ -0,0 +1,234 @@
+        titlefont_size=16,
+        showlegend=True,
+        hovermode="closest",
+        margin={"b": 20, "l": 5, "r": 5, "t": 40},
+        annotations=[annotations_style],
+        legend={"x": 0, "y": 1},
+        xaxis=plotly_axis_config,
+        yaxis=plotly_axis_config,
+    )
+    fig = go.Figure(data=[edge_trace, source_node_trace, target_node_trace], layout=layout)
+    fig.update_geos(fitbounds="locations")
+
+    if out_file is not None:
+        fig.write_html(out_file)
+    else:
+        fig.show()
+
+
+def plot_isolated_nodes(graph: HeteroData, out_file: Optional[Union[str, Path]] = None) -> None:
+    """Plot isolated nodes.
+
+    This method creates an interactive visualization of the isolated nodes in the graph.
+
+    Parameters
+    ----------
+    graph : AnemoiGraph
+        The graph to plot.
+    out_file : str | Path, optional
+        Name of the file to save the plot. Default is None.
+    """
+    isolated_nodes = compute_isolated_nodes(graph)
+
+    if len(isolated_nodes) == 0:
+        LOGGER.warning("No orphan nodes found.")
+        return
+
+    colorbar = plt.cm.rainbow(np.linspace(0, 1, len(isolated_nodes)))
+    nodes = []
+    for name, (lat, lon) in isolated_nodes.items():
+        nodes.append(
+            go.Scattergeo(
+                lat=lat,
+                lon=lon,
+                mode="markers",
+                hoverinfo="text",
+                name=name,
+                marker={"showscale": False, "color": colorbar[len(nodes)], "size": 10},
+            ),
+        )
+
+    layout = go.Layout(
+        title="<br>Orphan nodes",
+        titlefont_size=16,
+        showlegend=True,
+        hovermode="closest",
+        margin={"b": 20, "l": 5, "r": 5, "t": 40},
+        annotations=[annotations_style],
+        legend={"x": 0, "y": 1},
+        xaxis=plotly_axis_config,
+        yaxis=plotly_axis_config,
+    )
+    fig = go.Figure(data=nodes, layout=layout)
+    fig.update_geos(fitbounds="locations")
+
+    if out_file is not None:
+        fig.write_html(out_file)
+    else:
+        fig.show()
+
+
+def plot_interactive_nodes(graph: HeteroData, nodes_name: str, out_file: Optional[str] = None) -> None:
+    """Plot nodes.
+
+    This method creates an interactive visualization of a set of nodes.
+
+    Parameters
+    ----------
+    graph : HeteroData
+        Graph.
+    nodes_name : str
+        Name of the nodes to plot.
+    out_file : str, optional
+        Name of the file to save the plot. Default is None.
+    """
+    node_latitudes = np.rad2deg(graph[nodes_name].x[:, 0].numpy())
+    node_longitudes = np.rad2deg(graph[nodes_name].x[:, 1].numpy())
+    node_attrs = graph[nodes_name].node_attrs()
+    node_attrs.remove("x")
+
+    node_traces = {}
+    for node_attr in node_attrs:
+        node_attr_values = graph[nodes_name][node_attr].float().numpy()
+        if node_attr_values.ndim > 1 and node_attr_values.shape[1] > 1:
```

Comment:
what is this check for?

Reply:
Comment added. Only 1D attributes are represented in the interactive plots.